### PR TITLE
Admin permissions

### PR DESCRIPTION
**Pull request by bak45247**

### CODE REVIEW

File: tinker/admin/blink_roles/__init__.py [hunk 1]
```diff
@@ -11,8 +11,7 @@ class BlinkRolesView(FlaskView):
     route_base = '/admin/blink-roles'
 
     def before_request(self, args):
-        if 'Administrators' not in session['groups']:
-            abort(403)
+        admin_permissions('route_base', self, args)
```
Comment (c-schwarze) [inline tinker/admin/blink_roles/__init__.py]: should the 'route_base' string be the variable route_base?

(there are a bunch of these throughout the files

File: tinker/admin/program_search/__init__.py [hunk 2]
```diff
@@ -23,9 +23,7 @@ def __init__(self):
         self.base = ProgramSearchController()
 
     def before_request(self, args):
-        # give access to admins and lauren
-        if 'Administrators' not in session['groups'] and 'parlau' not in session['groups'] and session['username'] != 'kaj66635':
-            abort(403)
+        admin_permissions(self, app.config['ADMIN_PROGRAM_SEARCH_ROUTE_BASE'], args)
```
Comment (c-schwarze) [inline tinker/admin/program_search/__init__.py]: why is this one using configs?

File: tinker/tinker_controller.py [hunk 3]
```diff
@@ -86,6 +87,29 @@ def decorated(*args, **kwargs):
 #         return False
 
 
+# *args because some of the menus are called with args and some with **kwargs
+def admin_permissions(self, route_base, *args):
+    # this handles everything but redirects and the program search admin menus because they require extra permissions
+    if route_base != '/admin/redirect' and route_base != '/admin/program-search':
```
Comment (c-schwarze) [inline tinker/tinker_controller.py]: 1) should these if blocks include elif's.
2) you do a - if not /admin/redirect. But the next if block does a - if == /admin/redirect. That seems duplicated
3) should these be direct matches? or should they use 'in' i.e: if route_base in '/admin/redirect'

File: tinker/admin/blink_roles/__init__.py (hunk 1 above)
Comment (bak45247) [inline tinker/admin/blink_roles/__init__.py]: yes it should be the variable

File: tinker/tinker_controller.py (hunk 3 above)
Comment (bak45247) [inline tinker/tinker_controller.py]: i fixed this to be one if-elif-else statement

Comment (bak45247) [inline tinker/tinker_controller.py]: also should be direct matches

File: tinker/admin/program_search/__init__.py (hunk 2 above)
Comment (bak45247) [inline tinker/admin/program_search/__init__.py]: also switched this to be simpler

**Pull request merged**